### ISSUE TITLE
Fingerbank 500 Can't connect Sslv3 (on CentOS 8) 

### ISSUE THREAD

**Issue by oguzhantanir** [inverse-inc/packetfence#6728]
Hi Julsemaan,

I get the following error;
"
- Nov 30 13:14:26 XXX packetfence_httpd.portal[3162]: httpd.portal(3162) ERROR: [mac:00:11:22:33:44:55] Error while communicating with the Fingerbank collector. 500 Can't connect to 127.0.0.1:4723 (SSL connect attempt failed error:1408F10B:SSL routines:ssl3_get_record:wrong version number) (pf::fingerbank::update_collector_endpoint_data)
- Nov 30 13:14:26 XXX pfqueue[2351]: pfqueue(2351) ERROR: [mac:unknown] Error while communicating with the Fingerbank collector. 500 Can't connect to 127.0.0.1:4723 (SSL connect attempt failed error:1408F10B:SSL routines:ssl3_get_record:wrong version number) (pf::fingerbank::endpoint_attributes)
- Nov 30 13:14:26 XXX pfqueue[2351]: pfqueue(2351) ERROR: [mac:unknown] Unable to fetch query arguments for Fingerbank query. Aborting. (pf::fingerbank::process)

Packetfence 11.1 is connecting api.fingerbank.com and getting update, that is fine. We do not use proxy.
Where is the problem, what is your comment?

Thank you
Oguzhan

**Comment by julsemaan** [comment]
This uses the same certificate as your admin+portal. Could you post the following:
```
openssl x509 -text -in /usr/local/pf/conf/ssl/server.crt
```

**Comment by oguzhantanir** [comment]
- I did a clean install, I did not do any ssl creation.

[root@xxx ~]# openssl x509 -text -in /usr/local/pf/conf/ssl/server.crt
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            4d:37:8f:52:98:8d:01:64:bb:52:d7:ae:82:0b:27:3e:33:43:fb:fb
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = CA, ST = QC, L = Montreal, O = Inverse, CN = 127.0.0.1, emailAddress = support@inverse.ca
        Validity
            Not Before: Nov 25 10:10:58 2021 GMT
            Not After : Nov 25 10:10:58 2022 GMT
        Subject: C = CA, ST = QC, L = Montreal, O = Inverse, CN = 127.0.0.1, emailAddress = support@inverse.ca
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:e5:be:73:42:54:d7:30:ec:04:c2:34:80:9f:f5:
                    e7:82:62:fa:ae:d0:e4:a3:89:fc:58:3f:be:c0:82:
                    6d:b9:1e:71:27:e7:1a:f8:56:06:15:4b:cb:f0:13:
                    ba:e8:70:91:cf:72:f9:e0:32:65:ff:d0:f4:a3:49:
                    46:e3:b7:5f:24:0d:af:4b:8c:2b:2f:d8:cb:95:be:
                    75:f2:75:ac:67:6c:f9:f5:9c:71:cb:63:6a:2c:4f:
                    cd:2c:50:09:c4:9d:20:94:29:b7:03:2d:a3:4b:1c:
                    87:4a:ff:e8:89:4f:87:65:33:b8:34:08:14:16:ae:
                    84:09:f5:b0:db:da:67:36:2e:77:55:68:b4:a9:49:
                    01:7f:d6:9c:e7:8a:b3:a1:0c:dc:c1:63:18:22:e0:
                    80:8b:ae:ac:6e:81:0f:56:41:27:8b:4c:68:a0:f6:
                    b9:b3:9a:f9:5d:6c:ad:d7:dc:bf:ec:06:9a:57:9d:
                    4a:4c:6c:e7:56:b2:30:6a:9c:39:46:41:a7:31:f7:
                    7f:42:e5:45:5f:8e:9a:43:28:2d:f9:83:46:8a:34:
                    47:d9:6c:db:f2:59:55:89:c0:6e:2c:52:3c:87:42:
                    f1:63:9c:77:66:d5:e5:03:79:9f:38:e9:57:3f:33:
                    83:6d:0c:6a:84:33:75:74:02:f0:f3:51:52:95:2d:
                    38:e5
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Subject Key Identifier:
                85:D7:7D:E4:FE:52:A9:21:E2:01:C8:52:07:72:EA:5A:83:85:4A:CA
            X509v3 Authority Key Identifier:
                keyid:85:D7:7D:E4:FE:52:A9:21:E2:01:C8:52:07:72:EA:5A:83:85:4A:CA
                DirName:/C=CA/ST=QC/L=Montreal/O=Inverse/CN=127.0.0.1/emailAddress=support@inverse.ca
                serial:4D:37:8F:52:98:8D:01:64:BB:52:D7:AE:82:0B:27:3E:33:43:FB:FB

            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Issuer Alternative Name:
                <EMPTY>

            X509v3 Key Usage:
                Digital Signature, Non Repudiation, Key Encipherment, Certificate Sign
            X509v3 Extended Key Usage:
                TLS Web Server Authentication
            Netscape Cert Type:
                SSL Server
            X509v3 Subject Alternative Name:
                DNS:*, DNS:192.0.2.1
    Signature Algorithm: sha256WithRSAEncryption
         39:fe:87:85:a0:77:34:cf:7f:08:72:b2:81:96:cd:d0:88:a7:
         36:d3:49:ae:f1:39:86:42:4f:24:d2:2b:ad:15:83:74:07:01:
         40:73:8b:5a:cb:46:68:05:08:52:7a:a8:6f:9d:8e:39:07:c3:
         6f:18:f7:97:e5:f8:6f:a3:66:91:be:b2:e2:39:1a:ef:e6:2d:
         2f:b6:01:6a:a0:7a:56:a8:6f:b2:ff:3d:a1:91:0c:bc:63:ed:
         2f:f9:2a:07:5e:32:db:d3:f4:05:57:ea:a7:84:69:8b:80:ed:
         86:3f:e6:27:e7:f9:68:4f:e6:db:42:eb:33:b0:62:f5:f3:cd:
         0f:28:40:af:a3:39:bc:40:ef:b3:ea:8a:2d:bb:72:fb:c2:4d:
         d8:b1:ac:b8:ee:31:20:7c:a0:0b:5d:13:39:66:21:3a:f9:a3:
         9c:8a:c2:6a:b7:3c:9b:5f:6b:3f:94:16:2f:a3:8d:f1:b7:f6:
         e8:65:2b:6b:7e:e3:43:92:34:45:ae:cc:2e:ac:2a:18:3d:ca:
         44:d3:d0:b2:64:d2:28:fb:2d:9a:cc:66:1f:e9:a4:e6:40:a9:
         ca:b5:f5:9c
-----BEGIN CERTIFICATE-----
MIIEvDCCA6SgAwIBAgIUTTePUpiNAWS7UteuggsnPjND+/swDQYJKoZIhvcNAQEL
BQAwdjELMAkGA1UEBhMCQ0ExCzAJBgNVBAgTAlFDMREwDwYDVQQHEwhNb250cmVh
bDEQMA4GA1UEChMHSW52ZXJzZTESMBAGA1UEAxMJMTI3LjAuMC4xMSEwHwYJKoZI
hvcNAQkBFhJzdXBwb3J0QGludmVyc2UuY2EwHhcNMjExMTI1MTAxMDU4WhcNMjIx
MTI1MTAxMDU4WjB2MQswCQYDVQQGEwJDQTELMAkGA1UECBMCUUMxETAPBgNVBAcT
CE1vbnRyZWFsMRAwDgYDVQQKEwdJbnZlcnNlMRIwEAYDVQQDEwkxMjcuMC4wLjEx
ITAfBgkqhkiG9w0BCQEWEnN1cHBvcnRAaW52ZXJzZS5jYTCCASIwDQYJKoZIhvcN
AQEBBQADggEPADCCAQoCggEBAOW+c0JU1zDsBMI0gJ/154Ji+q7Q5KOJ/Fg/vsCC
bbkecSfnGvhWBhVLy/ATuuhwkc9y+eAyZf/Q9KNJRuO3XyQNr0uMKy/Yy5W+dfJ1
rGds+fWccctjaixPzSxQCcSdIJQptwMto0sch0r/6IlPh2UzuDQIFBauhAn1sNva
ZzYud1VotKlJAX/WnOeKs6EM3MFjGCLggIuurG6BD1ZBJ4tMaKD2ubOa+V1srdfc
v+wGmledSkxs51ayMGqcOUZBpzH3f0LlRV+OmkMoLfmDRoo0R9ls2/JZVYnAbixS
PIdC8WOcd2bV5QN5nzjpVz8zg20MaoQzdXQC8PNRUpUtOOUCAwEAAaOCAUAwggE8
MB0GA1UdDgQWBBSF133k/lKpIeIByFIHcupag4VKyjCBswYDVR0jBIGrMIGogBSF
Of6HhaB3NM9/CHKygZbN0IinNtNJrvE5hkJPJNIrrRWDdAcBQHOLWstGaAUIUnqo
b52OOQfDbxj3l+X4b6Nmkb6y4jka7+YtL7YBaqB6Vqhvsv89oZEMvGPtL/kqB14y
29P0BVfqp4Rpi4Dthj/mJ+f5aE/m20LrM7Bi9fPNMylu8L7q5CDaqdiwwRrLu+ar
0birYWcBFveb/z+hZkc3ki2gDyhAr6M5vEDvs+qKLbty+8JN2LGsuO4xIHygC10T
OWYhOvmjnIrCarc8m19rP5QWL6ON8bf26GUra37jQ5I0Ra7MLqwqGD3KRNPQsmTS
KPstmsxmH+mk5kCpyrX1nA==
-----END CERTIFICATE-----

**Comment by julsemaan** [comment]
Can you provide me the version of the fingerbank-collector package on your instance

**Comment by oguzhantanir** [comment]
[root@xxx pf]# rpm -qa | grep fingerbank
fingerbank-4.2.2-20210720065749.339834741.0008.v4.2.2.el8.noarch
fingerbank-collector-1.3.18-1.el8.x86_64

**Comment by oguzhantanir** [comment]
The version seems up to date. Could it be Centos 8 doesn't support sslv3 for security reasons?
How can I bypass this on PF?

**Comment by julsemaan** [comment]
It does look like the SSL library on CentOS 8 is outdated.

We technically don't support CentOS 8 anymore (only RHEL 8) so let me try this in our lab against RHEL 8 and report my findings. If it works on RHEL 8 and not CentOS 8, we'll have to see if the team wishes to invest time to fix CentOS 8 related bugs

**Comment by julsemaan** [comment]
I can confirm this happens on RHEL8 as well.

I'll investigate and report.

In the meanwhile you can downgrade to fingerbank-collector-1.3.17 using:
```
yum downgrade fingerbank-collector-1.3.17 --enablerepo=packetfence
```

1.3.17 has worked fine in my test

**Comment by oguzhantanir** [comment]
I'm doing a clean install again, more carefully. ( PF 11.1 and FB-C 1.3.17 ) on RHEL8
I will post the result here.

**Comment by julsemaan** [comment]
I just pushed 1.3.19 with the fix in the repos.

The issue was with a recent fix for #6698 that didn't load the certificate+key and left the collector operating in plain HTTP

I'll close this, feel free to post again if 1.3.19 doesn't fix it for you

**Comment by oguzhantanir** [comment]
I upgraded 1.3.19 and the problem was fixed.
Thank you julsemaan.
